pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -145,6 +145,18 @@ func (ps Params) extractParamArrayLengths() map[string]int {
 	return arrayParamsLengths
 }
 
+// validateDuplicateParameters checks if a parameter with the same name is defined more than once
+func (ps Params) validateDuplicateParameters() (errs *apis.FieldError) {
+	taskParamNames := sets.NewString()
+	for i, param := range ps {
+		if taskParamNames.Has(param.Name) {
+			errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("params names must be unique, the same param: %s is defined multiple times at", param.Name), fmt.Sprintf("params[%d].name", i)))
```

Comment:
consider using [`ErrMultipleOneOf`](https://pkg.go.dev/knative.dev/pkg/apis#ErrMultipleOneOf) instead of `ErrGeneric`

Reply:
thanks @jerop, yup I had initially thought of implementing this with `ErrMultipleOneof` back then but its ambiguous. `ErrMultipleOneof` produces error message `"expected exactly one, got both"` which implies `expected exactly one, got both: pipelineRef, pipelineSpec`. Also, the same params could be defined more than twice and such error message could be confusing.

If you still think  `ErrMultipleOneof` is a better fit, I am more than happy to change it.

Reply:
makes sense, thanks for the explanation!